### PR TITLE
Added ability to pull profile icon and heading when present

### PR DESCRIPTION
**Pull request by cyrusae**
See #10

### CODE REVIEW

File: src/utils/user.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ const dds = $userProfile(".meta dd:not(.email):not(dt.location+dd):not(.birthday):not(.pseuds)").text();
+ return dds.slice(10);
+}
+
+export const getProfilePic = ($userProfile: UserProfile) => {
+ return $userProfile("img.icon").attr("src");
+}
+
+export const getProfileHeader = ($userProfile: UserProfile) => {
+ return $userProfile("h3.heading").text().trim().slice(15, -9);
```
[the remark this answers is not in the document]
Comment (cyrusae) [reply src/utils/user.ts:57]: Oh that makes things much simpler compared to me hand-counting the letters, goodness

File: src/utils/user.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  ) as UserProfile;
+};
+
+export const getProfileName = ($userProfile: UserProfile) => {
+ return $userProfile(".user.profile .header h2").text().trim();
+} 
+
+export const getProfilePseuds = ($userProfile: UserProfile) => {
+ const pseuds = $userProfile("dd.pseuds").text().concat(", ");
+ return pseuds.slice(0, -2);
```
[the remark this answers is not in the document]
Comment (cyrusae) [reply src/utils/user.ts]: Should be done; also found a more elegant way of resolving the heading anyway

[the remark this answers is not in the document]
Comment (cyrusae) [reply src/utils/user.ts]: Also fixed on the other branch that builds on this one

[the remark this answers is not in the document]
Comment (cyrusae) [reply src/utils/user.ts]: Added this explanation because it was apparently missing

File: src/utils/user.ts [hunk 3]
```diff
@@ -33,28 +33,34 @@ export const getProfileName = ($userProfile: UserProfile) => {
  return $userProfile(".user.profile .header h2").text().trim();
 } 
 
+//Trim punctuation; this allows us to remove the ", " between pseuds.
+const PSEUD_AFTER = ", ";
```
Comment (essential-randomness) [inline src/utils/user.ts:37]: use SUFFIX rather than AFTER

File: src/utils/user.ts [hunk 4]
```diff
@@ -33,28 +33,34 @@ export const getProfileName = ($userProfile: UserProfile) => {
...
 export const getProfilePseuds = ($userProfile: UserProfile) => {
  const pseuds = $userProfile("dd.pseuds").text().concat(", ");
- return pseuds.slice(0, -2);
+ return pseuds.slice(0, -PSEUD_AFTER.length);
 }
-// slice here prevents pseuds from getting an extra ", " at the end 
 
+//Dates are ten characters long in the following format:
+const DATE_CONTENT = "0000-00-00";
```
Comment (essential-randomness) [inline src/utils/user.ts:44]: use FORMAT rather than CONTENT

**Review by essential-randomness — APPROVED** [review]

**Pull request merged**